pkg/logproto/compat_test.go

```diff
@@ -105,3 +106,63 @@ func BenchmarkFromLabelAdaptersToLabelsWithCopy(b *testing.B) {
 		FromLabelAdaptersToLabelsWithCopy(input)
 	}
 }
+
+func TestLegacySampleCompatibilityMarshalling(t *testing.T) {
+	ts := int64(1232132123)
+	val := 12345.12345
+	legacySample := LegacySample{Value: val, TimestampMs: ts}
+	got, err := json.Marshal(legacySample)
+	require.NoError(t, err)
+
+	legacyExpected := fmt.Sprintf("[%d.%d,\"%.5f\"]", ts/1000, 123, val)
```

Comment:
This weird string manipulation is necessary because since `cortexpb.Sample` has timestamp in Ms format, it divides the ts by 1000 during the marshalling, while keeping the remainder. Ex:
```
if ts=19191919, then marshal(ts)=ts/1000=19191.919
```
I preferred to use a hardcoded timestamp instead of `time.Now()` because of a corner case that might happen whenever a timestamp multiple of 1000 is taken. In such a case, Go would keep the zeroes at the right, while `json.Marshal` wouldn't, requiring additional parsing logic.